mysql-async/src/test/java/com/github/jasync/sql/db/mysql/util/URLParserSpec.kt

```diff
@@ -35,15 +35,16 @@ class URLParserSpec {
     fun `create a jdbc mysql connection with the available fields`() {
         val connectionUri = "jdbc:mysql://128.167.54.90:9987/my_database?user=john&password=doe"
 
+        val expected = Configuration(
```

Comment:
The tests changes because they test equality on the `Confuguration` object level. Parsing the connection string creates a new `Configuration` instance that has a new `StaticCredentialsProvider` instance defaultet, which does not equal the defaulted instance on the expected `Configuration`.

Reply:
So maybe have CredentialsProvider.equals compare user and password instead?

Reply:
I don't like to mess with object equality if it's not really necessary, so I opted for refactoring the tests to only assert the fields that are relevant to the test instead of comparing the full configuration object.
I find is is cleaner in general.
Hope this makes sense to you.

Reply:
it's convenient for many cases to compare configuration data class. so I think it's better to keep that property...

Also I think it should be a simple change: make StaticCredentialsProvider a data class.

Reply:
Making the `StaticCredentialsProvider` would fix this behavior for a very specific variant of `Configuration` objects, but it's not a general solution. Any other `CredentialsProvider` would probably break this transitive contract unless we enforce it on the interface, which I personally would not do.
If we want to preserve this property of the `Configuration`, it would probably be best to implement a custom `equals()` method that only covers the 'static' fields.

Reply:
Practically, I think most of the users will use `StaticCredentialsProvider` so although it's not a general solution I think it's a good compromise for convenience for the majority of usages.

Reply:
I think this might be obsolete.
By defaulting the credentials provider, we break another feature of the `Configuration` data class, which is the copy constructor. For example, `Configuration().copy(password = 'foobar')` will not currently update the static provider.
I switched the implementation around to a lazy defaulting of the provider, which also solve the issues with the tests as the initial default on the `data class` is now `null`

Reply:
@oshai What you think, can we move forward with this approach? :)

Reply:
sorry I missed that. I think it looks good to merge. Will do that soon unless you have anything to add to the PR

Reply:
No worries, we are not in a hurry after all :)

I think this is good to merge. If you can find the time, I'd appreciate it if you could also release a new version.